### DISCUSSION TITLE
Azure Login Issues with AtlasOS - "No Refresh Token Found" Error

### DISCUSSION THREAD

**Discussion by Entervive** [Atlas-OS/Atlas#1288] [Help & Support] [answered]
### **Hello everyone,**

I'm encountering an issue with Azure on my AtlasOS setup and am hoping for some guidance on which modifications might be causing this. When I attempt to access Azure on this machine, I receive the following error message:

```
Interaction required: No refresh token found in the cache. Please sign-in.
Additional details: no_tokens_found
```

### **Error Details:**

```
{
  "sessionId": "505792509ada46c1be96422aa5429baa",
  "errors": [
    {
      "errorMessage": "no_tokens_found: No refresh token found in the cache. Please sign-in.",
      "clientId": "c44b4083-3bb0-49c1-b47d-974e53cbdf3c",
      "scopes": ["https://management.core.windows.net//.default"]
    },
    {
      "errorMessage": "no_tokens_found: No refresh token found in the cache. Please sign-in.",
      "clientId": "c44b4083-3bb0-49c1-b47d-974e53cbdf3c",
      "scopes": ["c44b4083-3bb0-49c1-b47d-974e53cbdf3c/.default"]
    }
  ]
}
```

### **Additional Information**
I’m able to log in to Azure without any issues on a standard Windows machine on the same network.
I've already tried signing in again and clearing my browser cache, but the problem persists on AtlasOS. Here is how the error looks like in Azure.

![image](https://github.com/user-attachments/assets/6449cf78-d6f1-4218-a2ae-5d412f4a717c)

### **My Question**
Could anyone advise on which specific services or features I should re-enable to fix this issue? If anyone has managed to get Azure working on AtlasOS, I'd greatly appreciate any tips or a list of settings/services that should be adjusted.

Thank you in advance for your help!

**Comment by digbick000** [comment]
same issue with me

**Reply by Entervive** [reply]
check [the-P1neapple] response for me the issue was due to my computer's time not being synced correctly

**Comment by the-P1neapple** [comment, marked as answer]
Hello @Entervive,

Firstly, I want to acknowledge that this project was intended for personal use, and we do not recommend installing Atlas on machines for business use without knowing exactly what our playbooks do. If you wish to use our optimisations in a business environment, we recommend looking into our playbook source and manually applying optimisations through an MDM or equivalent. 

Can you please tell me what version of the playbook you are using? It can be found in Settings > System > About

I am assuming you are logging in from a browser? If so, can you please tell me what browser you are having the issue on?

Please follow these troubleshooting steps.
1. Clear your browser cache. Steps vary for each browser, but you generally go to your browser's settings, and an option to clear browser data will be under a privacy and security section.
2. Ensure your computer time is synced correctly.

**Reply by Entervive** [reply]
Hello,

Thank you for your response and the troubleshooting steps. I have checked and realized that the issue was due to my computer's time not being synced correctly, as the automatic synchronization was turned off.

I am using Atlas OS because, after an unexplained computer failure, the modifications provided by Atlas have helped it function normally again.

Thank you for your assistance.
